### PR TITLE
basic implementation of the Operation and CollaborationManager

### PR DESCRIPTION
**Pull request by nikmel2803**
- `Operation` class definition
- `CollaborationManager` class definition
- `CollaborationManager` can listen to the `EditorJSModel` but does nothing so far
- `CollaborationManager` has method `applyOperation` that can modify model by applying operations (only text insertions/deletions at the moment)
- a couple of small tests have been written

next:
- support more operations and add more tests
- undo/redo logic
- integration with playground

### CODE REVIEW

File: packages/collaboration-manager/src/CollaborationManager.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  public applyOperation(operation: Operation): void {
+    const { blockIndex, dataKey, textRange } = operation.index;
+
+    if (blockIndex == undefined || dataKey == undefined || textRange == undefined) {
+      throw new Error('Unsupported index');
+    }
+
+    switch (operation.type) {
+      case OperationType.Insert:
+        this.#model.insertText(blockIndex, dataKey, operation.data.newValue);
```
[the remark this answers is not in the document]
Comment (nikmel2803) [reply packages/collaboration-manager/src/CollaborationManager.ts:38]: > only text insertions/deletions at the moment

File: packages/collaboration-manager/src/Operation.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  /**
+   * Index in the document model tree
+   */
+  public index: Index;
+
+  /**
+   * Operation data
+   */
+  public data: OperationData;
```
[the remark this answers is not in the document]
Comment (nikmel2803) [reply packages/collaboration-manager/src/Operation.ts:45]: we better change this when it becomes more clear. it should work given current scope

**Review by gohabereg — APPROVED** [review]

File: packages/collaboration-manager/src/CollaborationManager.spec.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
+/* eslint-disable @typescript-eslint/no-magic-numbers */
+import { createDataKey, IndexBuilder } from '@editorjs/model';
+import { EditorJSModel } from '@editorjs/model';
+import { CollaborationManager } from './CollaborationManager.js';
+import { Operation, OperationType } from './Operation.js';
+
+describe('CollaborationManager', () => {
+  it('should add text', () => {
```
Comment (neSpecc) [inline packages/collaboration-manager/src/CollaborationManager.spec.ts:8]:
```suggestion
  it('should add text on apply Insert Operation', () => {
```

File: packages/collaboration-manager/src/CollaborationManager.spec.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            fragments: [],
+          },
+        },
+      } ],
+      properties: {},
+    });
+  });
+
+
+  it('should remove text', () => {
```
Comment (neSpecc) [inline packages/collaboration-manager/src/CollaborationManager.spec.ts:48]:
```suggestion
  it('should remove text on apply Remove Operation', () => {
```

**Review by neSpecc — APPROVED** [review]

**Pull request merged**